numpy/fft/_pocketfft.c

```diff
@@ -2281,14 +2299,23 @@ execute_real_forward(PyObject *a1, double fct)
     return (PyObject *)ret;
 }
 static PyObject *
-execute_real_backward(PyObject *a1, double fct)
+execute_real_backward(PyObject *a1, double fct, PyObject *out)
 {
     rfft_plan plan=NULL;
-    PyArrayObject *data = (PyArrayObject *)PyArray_FromAny(a1,
-            PyArray_DescrFromType(NPY_CDOUBLE), 1, 0,
-            NPY_ARRAY_DEFAULT | NPY_ARRAY_ENSUREARRAY | NPY_ARRAY_FORCECAST,
-            NULL);
-    if (!data) return NULL;
+    PyArrayObject *data;
+    if(out == Py_None) {
+      data = (PyArrayObject *)PyArray_FromAny(a1,
+              PyArray_DescrFromType(NPY_CDOUBLE), 1, 0,
+              NPY_ARRAY_DEFAULT | NPY_ARRAY_ENSUREARRAY | NPY_ARRAY_FORCECAST,
+              NULL);
+      if (!data) return NULL;
+    }
+    else {
+      data = (PyArrayObject*)PyArray_EnsureArray(out);
```

Comment:
In this branch, one needs to be sure the dtype and shape are both correct. Does `PyArray_CopyObject` take care of that?

Reply:
That's my understanding, yes.